component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateUtil.java

```diff
@@ -398,4 +401,33 @@ public static boolean isAccountLock(AuthenticatedUser user) {
         return accountLock;
     }
 
+    /**
+     * Check whether user account is disabled or not.
+     *
+     * @param user Authenticated user.
+     * @return boolean account disabled or not.
+     */
+    public static boolean isAccountDisabled(AuthenticatedUser user) {
+
+        boolean accountDisable = false;
+        if (user != null) {
+            int tenantId = IdentityTenantUtil.getTenantId(user.getTenantDomain());
+            RealmService realmService = X509CertificateDataHolder.getInstance().getRealmService();
+            UserStoreManager userStoreManager;
+            UserRealm userRealm;
+            try {
+                userRealm = realmService.getTenantUserRealm(tenantId);
+                userStoreManager = userRealm.getUserStoreManager();
+                Map<String, String> values = userStoreManager.getUserClaimValues(IdentityUtil.addDomainToName(user
+                        .getUserName(), user.getUserStoreDomain()), new String[]{
+                        X509CertificateConstants.ACCOUNT_DISABLED_CLAIM}, UserCoreConstants.DEFAULT_PROFILE);
+                accountDisable = Boolean.parseBoolean(values.get(X509CertificateConstants.ACCOUNT_DISABLED_CLAIM));
+            } catch (UserStoreException e) {
+                if (log.isDebugEnabled()) {
```

Comment:
Since this exception is silently bypassed, if the account disabled state could not returned due to some reason, user will be allowed to log in. Is that the intended behaviour?

Reply:
IMO **account disable** and **account lock** are additional conditions we are checking in the x509 authenticator thus if some issue comes we need to rely on the certificate validation mainly rather than failing the authentication. WDYT?

Reply:
Fixed with 7ab4943